apps/passport-client/src/backgroundJobs.ts

```diff
@@ -0,0 +1,225 @@
+     * will reschedule itself in the same way.
+     */
+    const jobPollServerUpdates = (): void => {
+      // Mark that poll has started.
+      console.log("[JOB] polling server for updates");
+      activePollTimeout = undefined;
+      try {
+        // Do the real work of the poll.
+        doPollServerUpdates();
+      } finally {
+        // Reschedule next poll.
+        lastBackgroundPoll = Date.now();
+        setupPolling();
+      }
+    };
+
+    const doPollServerUpdates = async (): Promise<void> => {
+      const state = getState();
+      if (
+        !state.self ||
+        !!state.userInvalid ||
+        !!state.anotherDeviceChangedPassword
+      ) {
+        console.log("[JOB] skipping poll with invalid user");
+        return;
+      }
+
+      // Check for updates to User object.
+      try {
+        await pollUser(state.self, dispatch);
+      } catch (e) {
+        console.log("[JOB] failed poll user", e);
+      }
+
+      // Trigger extra download from E2EE storage, and extra fetch of
+      // subscriptions, but only if the first-time sync had time to complete.
+      if (state.completedFirstSync) {
+        update({
+          extraDownloadRequested: true,
+          extraSubscriptionFetchRequested: true
+        });
+      }
+    };
+
+    const jobCheckConnectivity = async (): Promise<void> => {
+      window.addEventListener("offline", () => setIsOffline(true));
+      window.addEventListener("online", () => setIsOffline(false));
+    };
+
+    const setIsOffline = (offline: boolean): void => {
+      console.log(`[CONNECTIVITY] ${offline ? "offline" : "online"}`);
+      update({
+        offline: offline
+      });
+      if (offline) {
+        toast("Offline", {
+          icon: "❌",
+          style: {
+            width: "80vw"
+          }
+        });
+      } else {
+        toast("Back Online", {
+          icon: "👍",
+          style: {
+            width: "80vw"
+          }
+        });
+      }
+    };
+
+    const startJobSyncOfflineCheckins = async (): Promise<void> => {
+      await jobSyncPodboxOfflineCheckins();
+      setInterval(jobSyncPodboxOfflineCheckins, 1000 * 60);
+    };
+
+    const jobSyncPodboxOfflineCheckins = async (): Promise<void> => {
+      const state = getState();
+      if (!state.self || state.offline) {
+        return;
+      }
+
+      const credentialManager = new CredentialManager(
+        getState().identity,
+        getState().pcds,
+        getState().credentialCache
+      );
+
+      if (state.checkedInOfflinePodboxTickets.length > 0) {
+        const ticketsByEvent = state.checkedInOfflinePodboxTickets.reduce(
+          (res, current) => {
+            if (res[current.eventId]) {
+              res[current.eventId].push(current.id);
+            } else {
+              res[current.eventId] = [current.id];
+            }
+            return res;
+          },
+          {} as Record<string, string[]>
+        );
```

Comment:
This isn't equivalent to `_.groupBy`. That would give us a `Record` of type `<string, OfflineTicket[]>` rather than `<string, string[]>`. Where we do `              res[current.eventId] = [current.id]`, `_.groupBy` would do `res[current.eventId] = [current]`.